### PR TITLE
[CAS-804] Alert for big files

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
[CAS-804](https://stream-io.atlassian.net/browse/CAS-804)
### Description

Adding alert for big files.

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog updated with client-facing changes
- ~[ ] New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt [hunk 1]
```diff
@@ -206,6 +206,7 @@ public class MessageInputView : ConstraintLayout {
         binding.dismissInputMode.setOnClickListener { dismissInputMode(inputMode) }
         setMentionsEnabled(style.mentionsEnabled)
         setCommandsEnabled(style.commandsEnabled)
+        binding.messageInputFieldView.setAttachmentMaxFileMb(style.maxFileSize)
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt:209]: Why is it configurable? Are these limits extendable by our BE?

**Review by ogkuzmin — COMMENTED** [review]
It's good changes. It makes UX already better. But I think about better solution. Can we show limitations with a grid view? It let us reduce user steps to zero instead of one (choose file -> show alert). User will be able to see a warning before they choose any file

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt (hunk 1 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt:209]: The client can use its own backend, which can have different limits.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt [hunk 2]
```diff
@@ -50,6 +51,11 @@ public class MessageInputView : ConstraintLayout {
 
     private var onSendButtonClickListener: OnMessageSendButtonClickListener? = null
     private var typingListener: TypingListener? = null
+
+    public val attachmentBiggerThanMax: StateFlow<Boolean> by lazy {
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt:55]: Do not like to expose StateFlow in a view. We can pass just some simple handler like:
``` kotlin
fun interface BigFileSelectionHandler {
   fun handleBigFileSelected(fileSize: Int) 
}
```

It let us not to expose any state and don't use `StateFlow` that could be not so convenient for Java users.

**Comment by ogkuzmin** [comment]
We also can enhance UX of media gallery selection. We can disable for selection media files with size bigger than max size

**Review by ogkuzmin — APPROVED** [review]

**Pull request merged**